google-cloud-storage/src/test/java/com/google/cloud/storage/SerializationTest.java

```diff
@@ -67,14 +66,96 @@ public class SerializationTest extends BaseSerializationTest {
       Storage.BucketTargetOption.metagenerationNotMatch();
   private static final Map<StorageRpc.Option, ?> EMPTY_RPC_OPTIONS = ImmutableMap.of();
 
+  private static Storage STORAGE;
+  private static Blob BLOB;
+  private static Bucket BUCKET;
+  private static PageImpl<Blob> PAGE_RESULT;
+
+  @BeforeClass
+  public static void beforeClass() {
+    StorageOptions storageOptions =
+        StorageOptions.newBuilder()
+            .setProjectId("p")
+            .setCredentials(NoCredentials.getInstance())
+            .build();
+    STORAGE = storageOptions.getService();
+    BLOB = BLOB_INFO.asBlob(STORAGE);
+    BUCKET = BUCKET_INFO.asBucket(STORAGE);
+    PAGE_RESULT = new PageImpl<>(null, "c", Collections.singletonList(BLOB));
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    if (STORAGE != null) {
+      STORAGE.close();
```

Comment:
This will be a "breaking change" in behavior no when using gRPC or am I misunderstanding?
We need to call `#close()` to shut down gRPC but wasn't a requirement before.

Reply:
For existing http, `#close()` is a no-op. With the gRPC resources we need a way to tell it to shutdown otherwise they may leak. For gRPC itself, it has a leak detector and will provide a stack trace of where the allocation happened and later didn't cleanup.

This addition, actually also brings us closer to the gapics by being able to use try-with-resource for the client:

```
try (Storage s = StorageOptions.grpc().build().getService()) {
  Page<Bucket> buckets = s.list();
} // automatically calls #close to clean things up
```

Reply:
Gotcha, so no breaking change for HTTP. I misudnerstood. 

Thanks Ben.